### PR TITLE
Add github action for C++ `clang-format` style check 

### PR DESCRIPTION
**Pull request by yinqingh**
Fix https://github.com/NVIDIA/spark-rapids-jni/issues/202. 

This PR includes two changes:

- Reformat all files based on cudf `.clang-format` style
- Add github action for C++ `clang-format` style check

### CODE REVIEW

**Comment by pxLi** [comment]
Please help provide some test runs (like in your forked repos), to show both success and failure cases, thanks

**Comment by pxLi** [comment]
also cc @ttnghia to help check if the auto-formatted changes look good thanks

**Review by ttnghia — APPROVED** [review]

**Comment by pxLi** [comment]
seems we should have enough time to merge this into 23.12 to avoid auto-merge commits to override the format checks.

re-targeted it to branch-23.12

File: .github/workflows/clang-format.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow to check clang format
+name: clang format check
+
+on:
+  pull_request_target:
```
Comment (pxLi) [inline .github/workflows/clang-format.yml:19]: just saw you are not getting correct merged share with `pull_request_target + pre-commit/action`.
then try using pull_request directly

Comment (yinqingh) [reply .github/workflows/clang-format.yml:19]: Yes. Seems it always checkout the default branch in the tests. Will fix this. Thanks!

File: .github/workflows/clang-format.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow to check clang format
+name: clang format check
+
+on:
+  pull_request:
```
Comment (yinqingh) [inline .github/workflows/clang-format.yml:19]: Change to `pull_request ` event

**Comment by pxLi** [comment]
just caught a new merged commit format issue at https://github.com/NVIDIA/spark-rapids-jni/actions/runs/7014779963/job/19083019630?pr=1597 👍 

please upmerge and re-format to get a clean pass, we should be good to merge

**Comment by pxLi** [comment]
build

**Review by pxLi — APPROVED** [review]

**Comment by yinqingh** [comment]
Test clang check CI in the PR: https://github.com/yinqingh/spark-rapids-jni/pull/4
- https://github.com/yinqingh/spark-rapids-jni/actions/runs/7014753289/job/19082947944?pr=4
- https://github.com/yinqingh/spark-rapids-jni/actions/runs/7014769163/job/19082990980?pr=4

**Pull request merged**